mlfinlab/bet_sizing/kelly.py

```diff
@@ -0,0 +1,102 @@
+    """
+    Calculates optimal bet size based on the kelly criterion.
+
+    This method assumes our strategy is used for the environment which have discrete and binary outcomes for each trial
+    such as coin tossing and sports betting. It returns the optimal betting size based on the kelly criterion.
+
+    This method is designed to deal with not only the scenario that we loss all of wager when we loss(loss_unit=1)
+    but also the general scenario. For example, when betting odds are 2, we should set 'profit_unit' to 2 and
+    'loss_unit' to 1.
+
+    The implementation is based on the paper, 'The Kelly Criterion in blackjack, sports betting, and the stock market'
+    by Edward O. Thorp.
+
+    :param win_probability: (float) This is probability of a win of your strategy.
+    :param profit: (float) profit per unit bet.
+    :param loss: (float) loss per unit bet.
+    :return: (kelly_bet_size, expected_growth_rate) (tuple[float, float])
+            kelly_bet_size: (float) optimal bet size from kelly criterion.
+            expected_growth_rate: (float) (expected) growth rate of balance(capital) when you bet as much as kelly bet
+            size.
+    """
+    loss_probability = 1 - win_probability
+
+    kelly_bet_size = win_probability / loss_unit - loss_probability / profit_unit
+    expected_growth_rate = loss_probability * np.log(1 - loss_unit * kelly_bet_size) + \
+                           win_probability * np.log(1 + profit_unit * kelly_bet_size)
+
+    return kelly_bet_size, expected_growth_rate
+
+
+def kelly_investing(returns: np.ndarray, risk_free_rate: float, annualize_factor: int) -> Tuple[float, float]:
+    """
+    Calculates optimal leverage of investing on a single asset based on the kelly criterion.
+
+    This method assumes our strategy is used for a single asset in the environment where there is not a finite number
+    of outcomes like investing on securities in the stock market.
+
+    The implementation is based on the paper, 'The Kelly Criterion in blackjack, sports betting, and the stock market'
+    by Edward O. Thorp.
+
+    :param returns: (np.ndarray) This is series of returns of your strategy or a asset.
+    :param risk_free_rate: (float) this is the annualized return on the remaining capital which is not invested on your
+    main strategy. The rate of the risk free asset such as treasury bill rate is usually used.
+    :param annualize_factor: (int) It is a scale factor to estimate annualized return of the strategy which depends on
+    the period of returns.
+
+    :return: (kelly_leverage, expected_growth_rate) (tuple[float, float])
+            kelly_leverage: (float) optimal leverage size from kelly criterion.
+            expected_growth_rate: (float) (expected) growth rate of balance(capital) when you invest as much as optimal
+            leverage calculated from kelly criterion.
+    """
+    mean = np.mean(returns) * annualize_factor
+    var = np.var(returns, ddof=1) * annualize_factor
```

Comment:
Are you sure the annualized mean and variance are calculated right?

Reply:
For example, we think daily return as a random variable and we can estimate annual volatility like below.
volatility of annual return = sqrt(252) * daily volatility.

We use std as volatility. That means, 
std(annual return) = sqrt(252)*std(daily std)

So, If you square both sides, 
variance = 252 * variance
 
That's why I code like that.

Reply:
If we think daily return as a random variable, we can estimate like
E[annual return] = 252 * E[daily return]

Reply:
or Do you mean I should comment return should be log return?

Thanks

Reply:
If the function works with log returns it should be specified in the docstring. It would be also good if the possible input could be simple returns.

Reply:
I rethink this part. The kelly already assumes that our returns follow gaussian not log return is gaussian. I think I should clarify the assumption and keep the original code. What do you think? :)